File: app/src/main/java/com/hover/stax/languages/LanguageViewModel.java
package com.hover.stax.languages;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.hover.stax.R;
import com.yariksoffice.lingver.Lingver;

import java.util.ArrayList;
import java.util.List;

public class LanguageViewModel extends AndroidViewModel {
    private MutableLiveData<List<Lang>> languages;

    public LanguageViewModel(Application application) {
        super(application);
        languages = new MutableLiveData<>();
        languages.setValue(getLanguages());
    }

    public LiveData<List<Lang>> loadLanguages() {
        return languages;
    }

    private List<Lang> getLanguages() {
        String[] languageCodes = getApplication().getResources().getStringArray(R.array.supported_lang_codes);
        ArrayList<Lang> langs = new ArrayList<>();
        langs.add(new Lang(Lingver.getInstance().getLanguage()));
        for (String code : languageCodes) {
            Lang toAdd = new Lang(code);
            if (!langs.contains(toAdd))
                langs.add(toAdd);
        }
        return langs;
    }
}

File: app/src/main/java/com/hover/stax/languages/LanguageViewModel.kt
/*
 * Copyright 2022 Stax
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.hover.stax.languages

import android.app.Application
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.hover.stax.R
import com.yariksoffice.lingver.Lingver
import javax.inject.Inject

class LanguageViewModel @Inject constructor(
    val application: Application
) : ViewModel() {

    val languages = MutableLiveData<List<Lang>>()

    init {
        languages.value = getLanguages()
    }

    private fun getLanguages(): List<Lang> {
        val languageCodes = application.resources.getStringArray(R.array.supported_lang_codes)

        val langs = mutableListOf(Lang(Lingver.getInstance().getLanguage()))

        for (code in languageCodes) {
            val toAdd = Lang(code)
            if (!langs.contains(toAdd)) langs.add(toAdd)
        }

        return langs
    }
}